spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/batch/JobLauncherApplicationRunner.java

```diff
@@ -187,6 +187,7 @@ private void executeRegisteredJobs(JobParameters jobParameters) throws JobExecut
 				}
 				catch (NoSuchJobException ex) {
 					logger.debug(LogMessage.format("No job found in registry for job name: %s", jobName));
+					throw new JobExecutionException(ex.getMessage(), ex);
```

Comment:
`NoSuchJobException` is a `JobExecutionException` so I don't think there's any need to create a new exception. Instead, I think the `try` and `catch` could be removed completely so that the method just throws to original `NoSuchJobException`.

Reply:
Fixed in [this](https://github.com/spring-projects/spring-boot/pull/36039/commits/c4d7bef0ae3c202a2765c2a3dbfd626a069ff8e4) commit.